public/app/plugins/panel/canvas/Connections.tsx

```diff
@@ -116,11 +117,19 @@ export class Connections {
     this.connectionLine.setAttribute('x2', `${x}`);
     this.connectionLine.setAttribute('y2', `${y}`);
 
+    const connectionLineX1 = this.connectionLine.x1.baseVal.value;
+    const connectionLineY1 = this.connectionLine.y1.baseVal.value;
+    if (!this.didConnectionLeaveHighlight) {
+      const connectionLength = Math.hypot(x - connectionLineX1, y - connectionLineY1);
+      if (connectionLength > connectionAnchorHighlightOffset && this.connectionSVG) {
+        this.didConnectionLeaveHighlight = true;
+        this.connectionSVG.style.display = 'block';
+        this.isDrawingConnection = true;
+      }
+    }
+
```

Comment:
After mulling this over a few days I think I am comfortable with this approach 👍